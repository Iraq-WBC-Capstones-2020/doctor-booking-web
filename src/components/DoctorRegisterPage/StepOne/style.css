.formRow {
  margin-bottom: 50px !important;
}

.personal label {
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 24px;
  color: #000000;
}

.personal .form-control {
  margin-top: 5px;
  height: 48px;
  width: 425px;
  border: 2px solid #eaeaea;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  border-radius: 3px;
  font-size: 14px;
}

.personal .myCheck .form-check {
  display: -webkit-inline-box !important;
  display: -ms-inline-flexbox !important;
  display: inline-flex !important;
  margin-right: 20px;
}

.personal .myCheck .form-check label {
  margin-top: 27px !important;
  display: contents;
  font-size: 14px !important;
  font-weight: normal;
}

#photoUplod {
  margin-left: 35px;
  background-color: transparent;
  color: #007bff;
  border: 1px solid #007bff;
}
/*# sourceMappingURL=style.css.map */
