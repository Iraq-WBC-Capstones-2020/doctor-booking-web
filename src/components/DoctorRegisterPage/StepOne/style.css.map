{
    "version": 3,
    "mappings": "AAAA,AAAA,QAAQ,CAAC;EACP,aAAa,EAAE,eAAe;CAC/B;;AACD,AACE,SADO,CACP,KAAK,CAAC;EACJ,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,GAAG;EAChB,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;EACjB,KAAK,EAAE,OAAO;CACf;;AAPH,AASE,SATO,CASP,aAAa,CAAC;EACZ,UAAU,EAAE,GAAG;EACf,MAAM,EAAE,IAAI;EACZ,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,iBAAiB;EACzB,UAAU,EAAE,UAAU;EACtB,aAAa,EAAE,GAAG;EAClB,SAAS,EAAE,IAAI;CAChB;;AAjBH,AAoBI,SApBK,CAmBP,QAAQ,CACN,WAAW,CAAC;EACV,OAAO,EAAE,sBAAsB;EAC/B,YAAY,EAAE,IAAI;CAQnB;;AA9BL,AAwBM,SAxBG,CAmBP,QAAQ,CACN,WAAW,CAIT,KAAK,CAAC;EACJ,UAAU,EAAE,eAAe;EAC3B,OAAO,EAAE,QAAQ;EACjB,SAAS,EAAE,eAAe;EAC1B,WAAW,EAAE,MAAM;CACpB;;AAKP,AAAA,WAAW,CAAC;EACV,WAAW,EAAE,IAAI;EACjB,gBAAgB,EAAE,WAAW;EAC7B,KAAK,EAAE,OAAO;EACd,MAAM,EAAE,iBAAiB;CAC1B",
    "sources": [
        "style.scss"
    ],
    "names": [],
    "file": "style.css"
}