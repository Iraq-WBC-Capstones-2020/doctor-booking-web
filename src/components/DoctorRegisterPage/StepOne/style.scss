.formRow {
  margin-bottom: 50px !important;
}
.personal {
  label {
    font-style: normal;
    font-weight: 500;
    font-size: 18px;
    line-height: 24px;
    color: #000000;
  }

  .form-control {
    margin-top: 5px;
    height: 48px;
    width: 425px;
    border: 2px solid #eaeaea;
    box-sizing: border-box;
    border-radius: 3px;
    font-size: 14px;
  }

  .myCheck {
    .form-check {
      display: inline-flex !important;
      margin-right: 20px;

      label {
        margin-top: 27px !important;
        display: contents;
        font-size: 14px !important;
        font-weight: normal;
      }
    }
  }
}

#photoUplod {
  margin-left: 35px;
  background-color: transparent;
  color: #007bff;
  border: 1px solid #007bff;
}
