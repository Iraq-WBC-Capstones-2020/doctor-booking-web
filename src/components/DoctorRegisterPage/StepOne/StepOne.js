import React, { useState, useEffect, useContext } from 'react';
import './style.css';
import { DoctorContext, ACTIONS } from '../../../DoctorContext';
import { Form, Row, Col, Container } from 'react-bootstrap';

function StepOne() {
  const [state, dispatch] = useContext(DoctorContext);
  const [name, setName] = useState('');
  const [phoneNumber, setPhoneNumber] = useState('');
  const [gender, setGender] = useState('');
  const [password, setPassword] = useState('');
  const [email, setEmail] = useState('');
  const [img, setImg] = useState({});

  useEffect(() => {
    dispatch({
      type: ACTIONS.ADD_DOCTOR,
      doctorInfo: {
        name: name,
        phoneNumber: phoneNumber,
        gender: gender,
        password: password,
        email: email,
        img: img,
      },
    });
  }, [gender, name, password, phoneNumber, email, img]);
  useEffect(() => {
    setName(state.doctorInfo.name);
    setGender(state.doctorInfo.gender);
    setPassword(state.doctorInfo.password);
    setEmail(state.doctorInfo.email);
    setPhoneNumber(state.doctorInfo.phoneNumber);
  }, []);
  const HandleChange = (setField, e) => setField(e.target.value);
  const handleImg = (e) => {
    setImg(e.target.files[0]);
  };
  return (
    <Container>
      <div className="personal">
        <Row className="formRow">
          <Col lg={6}>
            <Form.Group>
              <Form.Label for="fullName">
                Full Name
                <Form.Control
                  name="fullName"
                  id="fullName"
                  placeholder="Full name"
                  onChange={(e) => HandleChange(setName, e)}
                  value={name}
                  required
                />
              </Form.Label>
            </Form.Group>
          </Col>
          <Col lg={6}>
            <fieldset>
              <Form.Group>
                <Form.Label name="formHorizontalRadios">
                  Gender
                  <div className="myCheck mt-2">
                    <Form.Check
                      type="radio"
                      label="female"
                      name="gender"
                      id="female"
                      value="female"
                      onChange={(e) => HandleChange(setGender, e)}
                      checked={gender === 'female' ? true : false}
                    />
                    <Form.Check
                      type="radio"
                      label="male"
                      name="gender"
                      id="male"
                      value="male"
                      onChange={(e) => HandleChange(setGender, e)}
                      checked={gender === 'male' ? true : false}
                    />
                  </div>
                </Form.Label>
              </Form.Group>
            </fieldset>
          </Col>
        </Row>
        <Row className="formRow">
          <Col lg={6}>
            <Form.Label for="E-mail">
              Email
              <Form.Control
                name="E-mail"
                id="E-mail"
                type="email"
                placeholder="Enter  your E-mail"
                onChange={(e) => HandleChange(setEmail, e)}
                value={email}
                required
              />
            </Form.Label>
          </Col>
          <Col lg={6}>
            <Form.Label for="number">
              Phone Number
              <Form.Control
                name="number"
                id="number"
                type="tel"
                placeholder="ex: 0770-145-6788"
                onChange={(e) => HandleChange(setPhoneNumber, e)}
                value={phoneNumber}
                required
              />
            </Form.Label>
          </Col>
        </Row>

        <Row className="formRow">
          <Col lg={6}>
            <Form.Label for="thePassword">
              Password
              <Form.Control
                name="thePassword"
                id="thePassword"
                type="password"
                placeholder="Enter your password"
                onChange={(e) => HandleChange(setPassword, e)}
                value={password}
                required
              />
            </Form.Label>
          </Col>
          <Col lg={6}>
            <Form.Label for="confirmPassword">
              Confirm Password
              <Form.Control
                id="confirmPassword"
                name="confirmPassword"
                type="password"
                placeholder="Enter your password again"
                required
              />
            </Form.Label>
          </Col>
        </Row>

        <Row className="formRow">
          <Col lg={12}>
            <label htmlFor="photoUplod">Personal Photo</label>
            <label className="btn btn-primary text-white mt-3 ml-5">
              Upload <input onChange={handleImg} type="file" hidden />
            </label>
          </Col>
        </Row>
      </div>
    </Container>
  );
}

export default StepOne;
