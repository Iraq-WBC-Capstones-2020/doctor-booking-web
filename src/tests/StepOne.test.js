import React from 'react';
import StepOne from '../components/DoctorRegisterPage/StepOne/StepOne';
import renderer from 'react-test-renderer';
import { renderHook } from '@testing-library/react-hooks';
import DoctorInfoProvider from '../DoctorContext';
import 'mutationobserver-shim';

it('Step one renders', () => {
  renderHook(() => <StepOne />);
});
it('Step 1 renders correctly', () => {
  const tree = renderer
    .create(
      <>
        <DoctorInfoProvider>
          <StepOne />
        </DoctorInfoProvider>
      </>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
