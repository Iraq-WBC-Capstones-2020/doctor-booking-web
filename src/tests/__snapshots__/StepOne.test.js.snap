// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step 1 renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="personal"
  >
    <div
      className="formRow row"
    >
      <div
        className="col-lg-6"
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
            for="fullName"
          >
            Full Name
            <input
              className="form-control"
              id="fullName"
              name="fullName"
              onChange={[Function]}
              placeholder="Full name"
              required={true}
              value=""
            />
          </label>
        </div>
      </div>
      <div
        className="col-lg-6"
      >
        <fieldset>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              name="formHorizontalRadios"
            >
              Gender
              <div
                className="myCheck mt-2"
              >
                <div
                  className="form-check"
                >
                  <input
                    checked={false}
                    className="form-check-input"
                    disabled={false}
                    id="female"
                    name="gender"
                    onChange={[Function]}
                    type="radio"
                    value="female"
                  />
                  <label
                    className="form-check-label"
                    htmlFor="female"
                    title=""
                  >
                    female
                  </label>
                </div>
                <div
                  className="form-check"
                >
                  <input
                    checked={false}
                    className="form-check-input"
                    disabled={false}
                    id="male"
                    name="gender"
                    onChange={[Function]}
                    type="radio"
                    value="male"
                  />
                  <label
                    className="form-check-label"
                    htmlFor="male"
                    title=""
                  >
                    male
                  </label>
                </div>
              </div>
            </label>
          </div>
        </fieldset>
      </div>
    </div>
    <div
      className="formRow row"
    >
      <div
        className="col-lg-6"
      >
        <label
          className="form-label"
          for="E-mail"
        >
          Email
          <input
            className="form-control"
            id="E-mail"
            name="E-mail"
            onChange={[Function]}
            placeholder="Enter  your E-mail"
            required={true}
            type="email"
            value=""
          />
        </label>
      </div>
      <div
        className="col-lg-6"
      >
        <label
          className="form-label"
          for="number"
        >
          Phone Number
          <input
            className="form-control"
            id="number"
            name="number"
            onChange={[Function]}
            placeholder="ex: 0770-145-6788"
            required={true}
            type="tel"
            value=""
          />
        </label>
      </div>
    </div>
    <div
      className="formRow row"
    >
      <div
        className="col-lg-6"
      >
        <label
          className="form-label"
          for="thePassword"
        >
          Password
          <input
            className="form-control"
            id="thePassword"
            name="thePassword"
            onChange={[Function]}
            placeholder="Enter your password"
            required={true}
            type="password"
            value=""
          />
        </label>
      </div>
      <div
        className="col-lg-6"
      >
        <label
          className="form-label"
          for="confirmPassword"
        >
          Confirm Password
          <input
            className="form-control"
            id="confirmPassword"
            name="confirmPassword"
            placeholder="Enter your password again"
            required={true}
            type="password"
          />
        </label>
      </div>
    </div>
    <div
      className="formRow row"
    >
      <div
        className="col-lg-12"
      >
        <label
          htmlFor="photoUplod"
        >
          Personal Photo
        </label>
        <label
          className="btn btn-primary text-white mt-3 ml-5"
        >
          Upload 
          <input
            hidden={true}
            onChange={[Function]}
            type="file"
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
